src/app/clusters/electrical-power-measurement-server/electrical-power-measurement-server.cpp

```diff
@@ -188,15 +190,25 @@ CHIP_ERROR Instance::Read(const ConcreteReadAttributePath & aPath, AttributeValu
         VerifyOrReturnError(
             HasFeature(ElectricalPowerMeasurement::Feature::kHarmonics), CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE,
             ChipLogError(Zcl, "Electrical Power Measurement: can not get HarmonicCurrents, feature is not supported"));
-        ReturnErrorOnFailure(
-            aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR { return this->EncodeHarmonicCurrents(encoder); }));
+        err = aEncoder.EncodeList([this](const auto & encoder) -> CHIP_ERROR { return this->EncodeHarmonicCurrents(encoder); });
+        if (CHIP_ERROR_NOT_FOUND == err)
+        {
+            ReturnErrorOnFailure(aEncoder.EncodeNull());
```

Comment:
Null and empty array are supposed to be semantically equivalent, no?  So anyone relying on the difference between the two is broken.